src/include/encoding.h

```diff
@@ -1439,7 +1439,7 @@ decode(std::array<T, N>& v, bufferlist::const_iterator& p)
   } while (false);							\
   if (struct_end) {							\
     if (bl.get_off() > struct_end)					\
-      throw buffer::malformed_input(DECODE_ERR_PAST(__PRETTY_FUNCTION__)); \
+      throw ::ceph::buffer::malformed_input(DECODE_ERR_PAST(__PRETTY_FUNCTION__)); \
```

Comment:
i wonder, if `buffer` is already in the same naming space of `decode()`, shall we guard `buffer` with `::ceph::`? i read libstdc++, libc++ and boost . none of them are referencing the class with the fully-qualified namespace prefix. what's your consideration of this change here?

Reply:
It's a macro. The body could be expanded in any scope, so I want to fully qualify it.